pydy/tests/test_joints.py

```diff
@@ -0,0 +1,275 @@
+
+        # 1. numeric value
+        self.joint.set_child_vel(10 * self.parent.frame.x)
+        assert self.child.masscenter.vel(self.parent.frame) == 10 * self.parent.frame.x
+
+        # 2. symbolic value
+        v1 = dynamicsymbols('v1')
+        self.joint.set_child_vel(v1 * self.parent.frame.x)
+        assert self.child.masscenter.vel(self.parent.frame) == v1 * self.parent.frame.x
+
+    def test_joint_set_joint_point_vel(self):
+        v2 = dynamicsymbols('v2')
+        self.joint.set_joint_point_vel(v2 * self.parent.frame.x)
+        assert self.joint.joint_point.vel(self.parent.frame) == v2 * self.parennt.frame.x
+
+    def test_joint_set_child_angular_velocity(self):
+        v3 = dynamicsymbols('v3')
+        self.joint.set_child_angular_vel(v3 * self.parent.frame.z)
+        assert self.child.frame.ang_vel_in(self.parent.frame) == v3 * self.parent.frame.z
+
+    def test_joint_orient_child(self):
+        q1 = Symbol('q1')
+        self.joint.orient_child('Axis', [q1, self.parent.frame.x])
+
+        # TODO other similar frame.orient() tests can be done.
+        # leaving more tests to discuss its necessity with others.
+
+
+class TestPinJoint():
+    def setup(self):
+        self.parent = Body('parent')
+        self.child = Body('child')
+        self.pinjoint = PinJoint('pinjoint', self.parent, self.child)
+
+    def test_pinjoint_init(self):
+        # default values
+        assert self.pinjoint.parent_axis == self.parent.frame.x
+        assert self.pinjoint.child_axis == self.child.frame.x
+        assert self.pinjoint.parent_point == self.parent.masscenter
+        assert self.pinjoint.child_point == self.child.masscenter
+        assert self.pinjoint.parent_joint_vector == self.parent.frame.x
+        assert self.pinjoint.child_joint_vector == self.child.frame.x
+
+        assert self.parent_point.vel(self.parent.frame) == 0
+        assert self.child_point.vel(self.child.frame) == 0
+
+    def test_pinjoint_parameters(self):
+        self.pinjoint = PinJoint('pinjoint', self.parent, self.child, par_point_vec_tuple=(1, 0, 0),
+                                 child_point_vec_tuple=(0, 1, 0), parent_axis='x', child_axis='y')
+        assert self.pinjoint.parent_axis == self.parent.frame.x
+        assert self.pinjoint.child_axis == self.child.frame.y
+        point1 = self.parent.masscenter.locatenew(self.parent.frame, self.parent.frame.x)
+        point2 = self.child.masscenter.locatenew(self.child.frame, self.child.frame.y)
+        assert self.pinjoint.parent_point == point1
+        assert self.pinjoint.child_joint_piont == point2
+        assert self.pinjoint.parent_joint_vector == self.parent.frame.x
+        assert self.pinjoint.child_joint_vector == self.chidl.frame.y
+
+    def test_pinjoint_tuple_vector(self):
+        a, b, c, d, e, f = symbols('a b c d e f')
+        pinjoint = PinJoint('pinjoint', self.parent, self.child, par_point_vec_tuple=(a, b, c),
```

Comment:
I thought about that and I have updated my implementation. This is common to other joints e.g. instead of velocity we can set a distance. i will update it after finishing what I am working on.